base/common/python/pki/nssdb.py

```diff
@@ -424,7 +424,16 @@ def modify_cert(self, nickname, trust_attributes):
         logger.debug('Command: %s', ' '.join(cmd))
         subprocess.check_call(cmd)
 
-    def create_noise(self, noise_file, size=2048):
+    def create_noise(self, noise_file, size=2048, key_type='rsa'):
+        # Under EC keys, key_size parameter is actually the name of a curve.
+        # This curve maps to a specific size, but EC keys require less entropy
+        # to generate than RSA keys. We can either maintain a mapping of
+        # curve name -> key size (and note that the openssl rand command takes
+        # the number of bytes, not the number of bits), or we can hard-code
+        # some safe value. We choose the latter.
+        if key_type.lower() in ('ec', 'ecc'):
+            size = 1024
+
```

Comment:
This will work just fine, but to keep it simple the `create_noise()` doesn't need to know the key type. The noise size can be computed based on the key type before calling this method. Please see my other comment.